osg-hosted-ce: create Certificate object if requested (SOFTWARE-6247)

matyasselmeci opened the pull request:
Adds the `HostCredentials.CreateCertificate` boolean -- if set to true (default false) then the chart will create a certmanager `Certificate` object using the issuer specified in `HostCredentials.CertificateIssuer` (default `letsencrypt-prod-newchain` which is the Let's Encrypt issuer on tiger).

The new certificate and resulting secret will be named after `HostCredentials.HostCertKeySecret` (if specified) or `<ce name>-cert` if not specified (which matches the existing pattern).

For backward compat, if `HostCredentials.LetsEncryptStaging` is true then it will use the `letsencrypt-staging` issuer instead, ignoring `HostCredentials.CertificateIssuer`. I'm wondering if that setting is still useful though...

matyasselmeci committed:
Add template for certmanager Certificate

matyasselmeci committed:
Bump Chart version to 4.15.0

brianhlin reviewed (changes requested):
As discussed in person, I would like the default cert + corresponding secret names to be different from the ones currently in production. This should make the transition easier for Ops

matyasselmeci committed:
Change default cert and secret name

brianhlin commented on supported/osg-htc/osg-hosted-ce/templates/certificate.yaml:
```diff
@@ -0,0 +1,27 @@
+    name: letsencrypt-staging
+    {{- else }}
+    name: letsencrypt-prod-newchain
+    {{- end }}
```
We should just let the operator set this since it's going to be custom per k8s cluster

brianhlin reviewed (approved):
LGTM except for one comment. Preapproving

matyasselmeci committed:
Drop LetsEncryptStaging; actually use HostCredentials.CertificateIssuer

Also add CertificateIssuerKind in case we have to use a regular Issuer
instead of a ClusterIssuer.
